### ISSUE TITLE
Default value for max_retries is too short to "clear" all rate limit windows

### ISSUE THREAD

**Issue by swarren** [python-gitlab/python-gitlab#1889]
## Description of the problem, including code/CLI snippet

The default value of max_retries is insufficient to "clear" any rate limiting limit, for responses where the Retry-After header is missing. The default is 10, which yields a max cur_retries value of 9, which yields a maximum sleep time of ~50 seconds. At least some rate limits are computed over a 10 minute window (600 seconds) (e.g. the recently added limit on /users/:id), so ideally we'd have a default max_retries that would (cumulatively) sleep at least that long.

## Expected Behavior

Default max_retries value is large enough to cause the library to sleep long enough to pass any rate limit window.

## Actual Behavior

The library doesn't sleep long enough, so a exception is thrown when the rate limit is reached, despite retrying for a while.

## Specifications

  - python-gitlab version: 1.15.0 (but I checked the latest code in git and it appears identical)
  - API version you are using (v3/v4): Whatever the library default is.
  - Gitlab server version (or gitlab.com): gitlab.com

**Comment by JohnVillalovos** [comment]
I notice this on the `user` endpoint

I removed the non `RateLimit` headers.

```
{'RateLimit-Limit': '2000',
 'RateLimit-Observed': '20',
 'RateLimit-Remaining': '1980',
 'RateLimit-Reset': '1645057248',
 'RateLimit-ResetTime': 'Thu, 17 Feb 2022 00:20:48 GMT'}
```

**Comment by JohnVillalovos** [comment]
Looks like we need to add support for these new headers: https://docs.gitlab.com/ee/user/admin_area/settings/user_and_ip_rate_limits.html#response-headers

**Comment by swarren** [comment]
@JohnVillalovos I don't believe that change will solve my original issue. There are no rate-limit or retry headers at all when the rate limit is hit on `/users/:id`:

`for i in $(seq 305); do curl --verbose --header "Authorization: Bearer xxx" "https://gitlab.com/api/v4/users/5400102"; done 2>&1 | tee log.txt
`

eventually yields:

`
HTTP/2 429 
< date: Thu, 10 Mar 2022 04:08:28 GMT
< content-type: application/json
< content-length: 89
< cache-control: no-cache
< vary: Origin
< x-content-type-options: nosniff
< x-frame-options: SAMEORIGIN
< x-request-id: 01FXS0728WE6KDCRW14TR5RE9P
< x-runtime: 0.026642
< strict-transport-security: max-age=31536000
< referrer-policy: strict-origin-when-cross-origin
< gitlab-lb: fe-01-lb-gprd
< gitlab-sv: api-gke-us-east1-c
< cf-cache-status: DYNAMIC
< expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
< report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v3?s=JfaflNapv18Po%2BZPk13XOp2%2FJJE0oSpTsz4CtpzSA3f0RBuphqWYa%2BWImkWAzxE5Urk20dQtug9veDTFkRFPqyCkK6WOr9Lzk7tTT7XbqrhnZBaZMFuj4pkKYcpaE7mhuySC8V%2BkDMI%3D"}],"group":"cf-nel","max_age":604800}
< nel: {"success_fraction":0.01,"report_to":"cf-nel","max_age":604800}
< server: cloudflare
< cf-ray: 6e99307aeccfc7b5-DEN
< 
{ [89 bytes data]
100    89  100    89    0     0    380      0 --:--:-- --:--:-- --:--:--   378
`

**Comment by JohnVillalovos** [comment]
> @JohnVillalovos I don't believe that change will solve my original issue. There are no rate-limit or retry headers at all when the rate limit is hit on `/users/:id`:
> 

Has an issue been filed with GitLab to figure out if this is "as intended" or an oversight on their part?

**Comment by swarren** [comment]
I haven't filed a bug

Gitlab support did notify me that the server-side changes would affect my usage, and I mentioned to them that python-gitlab should handle this based on rate limit headers so I wouldn't need to change anything. Then when the server-side change was rolled out I notified them again that python-gitlab wasn't able to handle this due to this missing rate limit headers. They didn't reply to that last point. Thus, at least part of gitlab should be aware of this, although I haven't formally filed a bug.

**Comment by bachsh** [comment]
Hello, I've also encountered this issue while working with gitlab.com API. It seems that the "Retry-After", etc. headers are missing, hence the default wait_time set by `python-gitlab` is too low.
IMO we should:
1. Set the default wait time should be 1 minute since gitlab.com rate limits are calculate per 1 minute.
2. Open a bug with gitlab's team.
